Source/WebCore/platform/graphics/cocoa/WebCoreDecompressionSession.mm

```diff
@@ -225,10 +223,12 @@
                     return;
                 }
                 assertIsCurrent(protectedThis->m_decompressionQueue.get());
-                if (!result)
+                if (protectedThis->m_lastDecodingError)
+                    producer.reject(protectedThis->m_lastDecodingError);
+                else if (!result)
                     producer.reject(kVTVideoDecoderBadDataErr);
                 else
-                    producer.resolve(std::exchange(protectedThis->m_lastDecodedSample, { }));
+                    producer.resolve(std::exchange(protectedThis->m_lastDecodedSamples, { }));
```

Comment:
Is there a case where result is false and m_lastDecodedSamples is not empty?
We could do the std::exchange unconditionally.

Reply:
> Is there a case where result is false and m_lastDecodedSamples is not empty? We could do the std::exchange unconditionally.

But then you can't return the error. 

m_lastDecodedSamples (and errors) will be cleared at the next call.

99.9% of the time, there's only ever one sample at play. The only exceptions possible today is vp8 with alternative frames, where the first frame returned would actually be null because it's not to be displayed